### DISCUSSION TITLE
How do you envision Sunbird Ed installation ?

### DISCUSSION THREAD

**Discussion by gandham-santhosh** [Sunbird-Ed/Community#432] [General Discussion]
Current installation process includes instructions to install, configure and deploying every component/microservice that is part of Ed. Having this approach is to enable an adopter to understand and leverage the generalisation and configurability features that are part of Ed so that one can customise suiting their use case and need.  

We would like to hear your thoughts on **HOW** do you envision Sunbird Ed installation? and **WHY**? with an example usecase/tool

**Comment by Krishnaj20** [comment]
@gandham-santhosh - Pls tag the respective community folks from whom you are seeking inputs.

**Reply by gandham-santhosh** [reply]
@saiakhil46 @santhosh-tg @tsprasath
@mathewjpallan @vrayulu @SanthoshVasabhaktula  @maheshkumargangula @anandp504 @reshmi-nair @vinukumar-vs

@AbhilashKD @abhishekb24 @adsingh7777 @agrawalanjali @Ashwiniev95 @asifbasha61 @BharathTarento @Debayanpaul1593 @Abhishekjain1106 @ankitpws @dhamoh @divyagovindaiah @Govind3877 @komalm @Priya-1123 @StartupSanatanaVenkatKL @Jyotiannoor @madhujaswal @manish050994 @manthanpandey23 @patilswati98 @pkranga @Premjith275 @ranvsin2 @Sanky-21 @Shakthieshwari @shiva6933 @snehal0904 @snehav05 @srajasimman @sudarshanswati @vatsaraj @vky2

Request your thoughts

**Comment by mathewjpallan** [comment]
It would be good to have 2 options for the installation

Quick setup - where all services and databases are pre-configured to run on Kubernetes along with flink jobs and leveraging MinioFS (running in the same env) as an S3 compatible cloud storage. Something like a single helm chart which has no external dependencies. This would be good for quick evaluation.

Pro/scale setup - where we continue to have the current set of jobs and configurability, but segregated by the building blocks so that anyone who wishes to install certain building blocks can leverage the same. 

@gandham-santhosh

**Reply by gandham-santhosh** [reply]
Thanks @mathewjpallan for your response. This is useful.
Using MinioFS is not in our scope right now. This will be a good option to try Sunbird in any kuberenetes cluster without depending on any cloud managed service including object store. Can you please help pointing to any applications leveraging MinIO ? And also from implementation stand point, does it require changes to be made to the micro-services connecting to object store and Cloudstore-sdk?

Request others to share your thoughts as well

**Reply by mathewjpallan** [reply]
Have come across MinioFS setup on VMs as the cloud storage provider in one of the sunbird adoptions on an on-prem datacenter. MinioFS is s3 compatible and so the cloud-storage-sdk just needs to be set with S3 configuration. This does not require changes in any services as the cloud-storage-sdk is compatible with s3/minofs

**Reply by amitbansal26** [reply]
Any thoughts on this can we do an on prem installation of sunbird ?
